core/src/main/scala/org/apache/spark/SparkException.scala

```diff
@@ -408,14 +432,17 @@ private[spark] class SparkSecurityException(
 private[spark] class SparkArrayIndexOutOfBoundsException(
     errorClass: String,
     errorSubClass: Option[String] = None,
-    messageParameters: Array[String],
+    messageParameters: Map[String, String],
```

Comment:
This is tested by QueryExecutionAnsiErrorsSuite "INVALID_ARRAY_INDEX: get element from array"